### ISSUE TITLE
CUPS unable to print using ipps:// with "The printer is not responding."

### ISSUE THREAD

**Issue by obadz** [NixOS/nixpkgs#315157]
### Describe the bug

Somewhere between 58a1abdba and 205781405, cups lost its ability to print using ipps:// end points (in my case an HP DeskJet 4100 printer).

Instead it flags the error "The printer is not responding".

Using the ipp:// endpoints still works fine.
Reverting to 58a1abdba works fine.

Reverting specific cups related commits 005483e2b83748174dc267890cd2cffa374df5de, 10f25e88898e078fad2969d371019f109fa51998, 07ec5ac79d1e29a3372f84d1eb6b41765ada4805 and 3edeb8fac849743833cb06981b66673d8e988e76 does **not** fix the problem.

### Steps To Reproduce

Print a document using CUPS on a printer configured with the ipps:// end point.

### Expected behavior

Document comes out of printer.

### Screenshots

```
$ journalctl -u cups.service -ef
...
cupsd[1344]: [Job 372] The printer is not responding.
cupsd[1344]: Saving subscriptions.conf...
cupsd[1344]: [Job 372] The printer is not responding.
cupsd[1344]: Saving subscriptions.conf...
...
```

### Additional context

I will try to add more information to this issue as it comes up.

### Notify maintainers

@matthewbauer 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.30, NixOS, 24.05 (Uakari), 24.05.git.pkgs-21b3130.conf-f18e0a0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `~/git/nixpkgs`
```

**Comment by obadz** [comment]
Nagging suspicion: broken by #306637 and fixed by #311735

**Comment by obadz** [comment]
Confirmed fixed in bfb7a8826
